sequencer/src/state.rs

```diff
@@ -130,43 +131,50 @@ async fn store_state_update(
         height = parent_leaf.height(),
     ),
 )]
-async fn update_state_storage(
+async fn update_state_storage<T>(
     parent_state: &ValidatedState,
-    storage: &Arc<RwLock<impl SequencerStateDataSource>>,
+    storage: &Arc<T>,
     instance: &NodeState,
     parent_leaf: &LeafQueryData<SeqTypes>,
     proposed_leaf: &LeafQueryData<SeqTypes>,
-) -> anyhow::Result<ValidatedState> {
+) -> anyhow::Result<ValidatedState>
+where
+    T: SequencerStateDataSource,
+    for<'a> T::Transaction<'a>: SequencerStateUpdate,
+{
     let parent_chain_config = parent_state.chain_config;
 
     let (state, delta) = compute_state_update(parent_state, instance, parent_leaf, proposed_leaf)
         .await
         .context("computing state update")?;
 
-    let mut storage = storage.write().await;
-    if let Err(err) = store_state_update(&mut *storage, proposed_leaf.height(), &state, delta).await
-    {
-        storage.revert().await;
```

Comment:
Shouldn't we keep it in case the transaction fails to commit?

Reply:
Oh I think it automatically reverts when the tx is dropped without being committed